aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -468,7 +468,7 @@ ilpReduce(index_t shift,
     if (offset >= shift && offset < size) {
       threadVal = r(threadVal, data[offset]);
     }
-    size -= blockDim.x;
+    size -= blockDim.x > size ? size : blockDim.x;
     data += blockDim.x;
```

Comment:
I think the size decrement should be same as data increment.

Reply:
This is basically just to prevent `size` from going negative when there is no-work to be done.
When the decrement is not the same as the data increment `size` is zero and the rest of the function becomes a no-op anyway.